pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -424,15 +478,59 @@ func (vm *windows) createService(svc *service) error {
 	return nil
 }
 
+// ensureServiceNotRunning stops a service if it exists and is running
+func (vm *windows) ensureServiceNotRunning(svc *service) error {
+	if svc == nil {
+		return errors.New("service object should not be nil")
+	}
+
+	exists, err := vm.serviceExists(svc.name)
+	if err != nil {
+		return errors.Wrap(err, "error checking if service exists")
+	}
+	if !exists {
+		// service does not exist, therefore it is not running
+		return nil
+	}
+
+	running, err := vm.isRunning(svc.name)
+	if err != nil {
+		return errors.Wrap(err, "unable to check if service is running")
+	}
+	if !running {
+		return nil
+	}
+	if err := vm.stopService(svc); err != nil {
+		return errors.Wrap(err, "unable to stop service")
+	}
+	return nil
+
+}
+
 // stopService stops the service that was already running
 func (vm *windows) stopService(svc *service) error {
 	if svc == nil {
 		return errors.New("service object should not be nil")
 	}
+	// Success here means that the stop has initiated, not necessarily completed
 	out, err := vm.Run("sc.exe stop "+svc.name, false)
 	if err != nil {
 		return errors.Wrapf(err, "failed to stop %s service with output: %s", svc.name, out)
 	}
+
+	// Wait until the service has stopped
+	err = wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+		serviceRunning, err := vm.isRunning(svc.name)
+		if err != nil {
+			log.V(1).Error(err, "unable to check if Windows service is running", "service", svc.name)
```

Comment:
error log is of the highest importance and I am not sure why we would want to use V(1) for it, if we want to have debug log we can use log.V(1).Info

Reply:
@mansikulkarni96 are you saying this `log.V(1).Error()` will show up on a release build? I think @sebsoto did not think this needs to be printed in release builds.